include/envoy/http/header_map.h

```diff
@@ -36,6 +40,7 @@ class LowerCaseString {
 
 private:
   void lower() { std::transform(string_.begin(), string_.end(), string_.begin(), tolower); }
+  bool valid() const { return string_.find('\0') == std::string::npos; }
```

Comment:
Can you add some kind of comment of why this indicates valid, why we are doing this, it's only used for asserts, etc.? (Same for the other valid function, or just have a comment there to point to here or vice versa)